src/enzyme_ad/jax/Passes/AffineToStableHLORaising.cpp

```diff
@@ -944,6 +978,217 @@ emitStoreAsScatter(Location loc, Value update, Value input, ValueRange sIndices,
   return res;
 }
 
+static LogicalResult tryRaisingForOpToStableHLOUnroll(
+    affine::AffineForOp forOp, IRMapping &mapping, OpBuilder &builder,
+    llvm::DenseMap<Value, affine::AffineValueMap> &maps, ParallelContext pc) {
+  auto tmpBlock = new Block();
+  OpBuilder oldFuncBuilder(builder.getContext());
+  oldFuncBuilder.setInsertionPointToStart(tmpBlock);
+  auto clonedFor = cast<affine::AffineForOp>(oldFuncBuilder.clone(*forOp));
+  auto yield = oldFuncBuilder.create<affine::AffineYieldOp>(
+      clonedFor.getLoc(), clonedFor.getResults());
+  if (failed(affine::loopUnrollFull(clonedFor))) {
+    delete tmpBlock;
+    return failure();
+  }
+  for (auto &innerOp : tmpBlock->without_terminator()) {
+
+    if (tryRaisingOpToStableHLO(&innerOp, mapping, builder, maps, pc).failed())
+      return failure();
+  }
+  for (auto [yielded, res] :
+       llvm::zip_equal(yield.getOperands(), forOp.getResults())) {
+    auto mapped = mapping.lookupOrNull(yielded);
```

Comment:
We very likely want to remove the mapping for `yielded` after this. The value will no longer exist after the block is deleted, but the mapping for it will persist, and may be erroneously associated with some other value that just happens to have the same address.

Maybe we can have a local IRMapping that we use for tmpBlock and then update that one with only the values that survive.

Reply:
I wanted to avoid the overhead of copying over the mapping but you are right that this could lead to some obscure bugs so I will do that.